src/Semantic/Api/Diffs.hs

```diff
@@ -81,20 +81,24 @@ diffGraph blobs = DiffTreeGraphResponse . V.fromList . toList <$> distributeFor
 
 type DiffEffects sig m = (Member (Error SomeException) sig, Member (Reader Config) sig, Member Telemetry sig, Member Distribute sig, Member Parse sig, Carrier sig m, MonadIO m)
 
-type Decorate a b = forall term diff . DiffActions term diff => Blob -> term a -> term b
 
+dotGraphDiffParsers :: Map Language (SomeParser DOTGraphDiff Loc)
+dotGraphDiffParsers = aLaCarteParsers
 
-class DOTGraphDiff diff where
-  dotGraphDiff :: (Carrier sig m, Member (Reader Config) sig) => diff Loc Loc -> m Builder
+class DiffTerms term => DOTGraphDiff term where
+  dotGraphDiff :: (Carrier sig m, Member (Reader Config) sig) => DiffFor term Loc Loc -> m Builder
 
-instance (ConstructorName syntax, Foldable syntax, Functor syntax) => DOTGraphDiff (Diff syntax) where
+instance (ConstructorName syntax, Diffable syntax, Eq1 syntax, Hashable1 syntax, Traversable syntax) => DOTGraphDiff (Term syntax) where
   dotGraphDiff = serialize (DOT (diffStyle "diffs")) . renderTreeGraph
 
 
-class JSONGraphDiff diff where
-  jsonGraphDiff :: BlobPair -> diff Loc Loc -> DiffTreeFileGraph
+jsonGraphDiffParsers :: Map Language (SomeParser JSONGraphDiff Loc)
+jsonGraphDiffParsers = aLaCarteParsers
```

Comment:
As documented in `Parsing.Parser`, it’s quite important that this be of concrete type, and thus a CAF: don’t return it in an abstract `Monad`, don’t generalize it over the constraint type. Otherwise your compile times go through the roof, as evinced by `-dshow-passes`.